File: backend/app/core/roles.py
from enum import StrEnum


class Role(StrEnum):
    ADMIN = 'admin'
    LIDER = 'lider'
    ESPECIALISTA = 'especialista'
    GESTOR = 'gestor'
    COLABORADOR = 'colaborador'


class MovementType(StrEnum):
    ENTRADA = 'entrada'
    SALIDA = 'salida'


MANAGE_ITEMS_ROLES = {Role.ADMIN, Role.LIDER, Role.ESPECIALISTA}
REPORT_ROLES = {Role.ADMIN, Role.LIDER, Role.ESPECIALISTA, Role.GESTOR}
ADMIN_ONLY = {Role.ADMIN}
ALL_ROLES = {Role.ADMIN, Role.LIDER, Role.ESPECIALISTA, Role.GESTOR, Role.COLABORADOR}
ALERT_RECIPIENT_ROLES = {Role.ADMIN, Role.LIDER, Role.ESPECIALISTA}

File: frontend/src/app/core/roles.ts
import { Role } from './api.types';

export const ROLE_LABELS: Record<Role, string> = {
  admin: 'Administrador',
  lider: 'Líder',
  especialista: 'Especialista',
  gestor: 'Gestor',
  colaborador: 'Colaborador'
};

export const ALL_ROLES: Role[] = ['admin', 'lider', 'especialista', 'gestor', 'colaborador'];
export const MANAGE_ITEM_ROLES: Role[] = ['admin', 'lider', 'especialista'];
export const REPORT_ROLES: Role[] = ['admin', 'lider', 'especialista', 'gestor'];
export const USER_ADMIN_ROLES: Role[] = ['admin'];

export function hasRole(userRole: Role | undefined, allowed: Role[]): boolean {
  return !!userRole && allowed.includes(userRole);
}
